pkg/parser/statement/statement_parser.go

```diff
@@ -185,70 +240,129 @@ func (sp *Parser) ParseValueSpec(spec *ast.ValueSpec) ([]*symboltable.SymbolDef,
 
 	var symbolsDef = make([]*symboltable.SymbolDef, 0)
 
+	// `var/const id, ... = expr` case:
 	if vLen == 1 {
+		// - parse `expr`:
 		valueExprAttr, err := sp.ExprParser.Parse(spec.Values[0])
 		if err != nil {
 			return nil, err
 		}
 
 		if builtin, ok := valueExprAttr.DataTypeList[0].(*gotypes.Builtin); !ok || builtin.Def != "iota" {
+			// Not the `iota` case:
+			// - `id1, id2, id3, ..., idn = expr` case (`expr` is probably a function call expression):
 			if nLen != len(valueExprAttr.DataTypeList) {
 				return nil, fmt.Errorf("ValueSpec %#v has different number of identifiers on LHS (%v) than a number of results of invocation on RHS (%v)", spec, nLen, len(valueExprAttr.DataTypeList))
 			}
+			// Iterate through identifiers:
 			for i, name := range spec.Names {
+				// - ignore the anonymous ones:
 				if name.Name == "_" {
 					continue
 				}
+				is_const := name.Obj.Kind == ast.Con
 				if typeDef == nil {
+					// Type not given; this symbol definition means that we are
+					// making a contract `var/const id = expr`; we must compute
+					// the type of `id` first:
+					// - in case of `var`, untyped "type" of `expr` became typed
+					//   according to Golang rules
+					// - in case of `const`, type is copied (if the assignemt is
+					//   possible)
+					t, err := resolveLhsType(valueExprAttr.DataTypeList[i], is_const)
+					if err != nil {
+						return nil, err
+					}
 					symbolsDef = append(symbolsDef, &symboltable.SymbolDef{
 						Name:     name.Name,
 						Package:  sp.PackageName,
-						Def:      valueExprAttr.DataTypeList[i],
-						Contract: valueExprAttr.Contract,
+						//Def:      valueExprAttr.DataTypeList[i],
+						Def: t, // resolved type
```

Comment:
IINM, this is a fix for an incorrect data type derivation. If so, it does not belong to the contracts. Please, create a separate PR for it.